pkgs/applications/networking/kubo-migrator/all-migrations.nix

```diff
@@ -23,8 +23,9 @@ let
   };
```

Comment:
use overrideAttrs to inherit all attributes, including `src` and `patches`

```nix
{
  fs-repo-common = pname: version: kubo-migrator-unwrapped.overrideAttrs (old: {
    inherit pname version;
    sourceRoot = "kubo-migrator-src/${pname}";
    vendorSha256 = null;
    doCheck = false;
    meta = old.meta // {
      mainProgram = pname;
      description = "Individual migration for the filesystem repository of Kubo clients";
    };
  });

```

Reply:
This doesn't work either because of `sourceRoot` as I already wrote in https://github.com/NixOS/nixpkgs/pull/209304#issuecomment-1402559385. Please try it yourself.

Reply:
> Please try it yourself.

good idea ^^

> go: inconsistent vendoring in /build/source/fs-repo-12-to-13

https://discourse.nixos.org/t/buildgomodule-with-local-src-inconsistent-vendoring/8641

> It looks like this is how a bad vendorSha256 manifests itself

https://discourse.nixos.org/t/inconsistent-vendoring-in-buildgomodule-when-overriding-source/9225/2

> overrideAttrs and buildGoModule aren’t aware of each other

https://github.com/NixOS/nixpkgs/issues/86349

&rarr; override buildGoModule
https://github.com/milahu/nixpkgs/tree/Luflosi/update/kubo-migrator

the problem is that overrideAttrs does not work, so `vendorHash != null`, so its using the wrong vendor folder

configurePhase:

```sh
runHook preConfigure

export GOCACHE=$TMPDIR/go-cache
export GOPATH="$TMPDIR/go"
export GOPROXY=off
export GOSUMDB=off
cd "$modRoot"
rm -rf vendor
cp -r --reflink=auto /nix/store/nd7scmc52v6bm2qvh7vwdqcd74a9bjzw-kubo-migrator-2.0.2-go-modules vendor


runHook postConfigure
```

see also
pkgs/build-support/go/module.nix

Reply:
This looks MUCH more complicated than what I'm doing, for seemingly no gain. Do you agree? Should this PR be merged as-is?

Reply:
looks complicated, because `buildGoModule.overrideGoAttrs` is missing in nixpkgs (also because i added lots of noise in my comments and drafts...)

but "usually" in nixpkgs, we create one base derivation, and from there, create multiple derivations using overrideAttrs

creating intermediary derivations (like youre doing) i would call an anti-pattern, because its a waste of disk space... but at the end of the day, it should "just work", so feel free to merge ; )

Reply:
Don't want to stop the idea here, but if I remember, the biggest problem in the migrator was that [qlts](https://github.com/ipfs/fs-repo-migrations/issues/157)
Wouldn't be easier try to upgrade this specific package to 1.18? Just for curiosity.

Reply:
> Wouldn't be easier try to upgrade this specific package to 1.18?

done in `patches`

Reply:
just read my reviews ...

my patches from https://github.com/ipfs/fs-repo-migrations/pull/163 can be reduced to something like

```nix
{
  # fix build on go 1.18: panic: qtls.ClientHelloInfo doesn't match
  # https://github.com/ipfs/fs-repo-migrations/pull/163
  patchPhase = ''
    substituteInPlace vendor/github.com/marten-seemann/qtls-go1-15/common.go \
      --replace \
        '"container/list"' \
        '"container/list"
        "context"' \
      --replace \
        'config *Config' \
        'config *Config
        ctx context.Context'
  '';
```

(i hope that go wont cry about the broken indents)

if you have more questions you can ask me on matrix `@milahu:matrix.org`

Reply:
Right, that's seems reasonable. Currently I'm using `steam-run` or `nix-alien` manually to fix this on my nodes, I'm thinking on using your branch for it (at least for now, because running this by hand is not a good practice).

Reply:
I updated this PR to use `substituteInPlace` as suggested above. This is much cleaner.